pkg/parcacol/querier.go

```diff
@@ -393,35 +403,17 @@ func (q *Querier) queryRangeDelta(
 		)
 	}
 
-	var perSecondExpr logicalplan.Expr
-	if resultType.Type == "cpu" && resultType.Unit == "nanoseconds" {
-		perSecondExpr = logicalplan.Div(
-			logicalplan.Convert(totalSum, arrow.PrimitiveTypes.Float64),
-			logicalplan.Convert(
-				logicalplan.If(
-					logicalplan.IsNull(timestampUnique),
-					logicalplan.Literal(step.Nanoseconds()),
-					durationMin,
-				),
-				arrow.PrimitiveTypes.Float64,
-			),
-		).Alias(ValuePerSecond)
-	} else {
-		perSecondExpr = logicalplan.Div(
```

Comment:
why was this branch removed?

Reply:
This entire block of code is executed within the queryRangeDelta function.
It only ever executes when we already checked for a delta profile:
```go
if queryParts.Delta {
	return q.queryRangeDelta(
		ctx,
		filterExpr,
		step,
		queryParts.Meta,
	)
}
```

Looking at this now, I think we need the check though as we can have delta profiles that aren't cpu/nanoseconds. So I'll bring this back.